src/Common/src/Microsoft/Win32/IMarshal.cs

```diff
@@ -0,0 +1,33 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Win32
+{
+    [ComImport, Guid("00000003-0000-0000-C000-000000000046"),
+    InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
+    internal interface IMarshal
+    {
+        [PreserveSig]
+        int GetUnmarshalClass(ref Guid riid, IntPtr pv, int dwDestContext, IntPtr pvDestContext, int mshlflags, out Guid pCid);
+        [PreserveSig]
+        int GetMarshalSizeMax(ref Guid riid, IntPtr pv, int dwDestContext, IntPtr pvDestContext, int mshlflags, out int pSize);
+        [PreserveSig]
+        int MarshalInterface(IntPtr pStm, ref Guid riid, IntPtr pv, int dwDestContext, IntPtr pvDestContext, int mshlflags);
+        [PreserveSig]
+        int UnmarshalInterface(IntPtr pStm, ref Guid riid, out IntPtr ppv);
+        [PreserveSig]
+        int ReleaseMarshalData(IntPtr pStm);
+        [PreserveSig]
+        int DisconnectObject(int dwReserved);
+    }
+
+    internal static class NativeMethods
```

Comment:
Not to be too dogmatic, but there is no way these constants aren't already defined somewhere. Is it possible to use them?